torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -374,6 +374,54 @@ Tensor TensorExprKernel::ComputeValue(torch::jit::Value* v) {
     case aten::pow: {
       return ComputeTwoOperand(
           "aten_pow", v, [](const Expr& lhs, const Expr& rhs) {
+            const FloatImm* float_imm = rhs.AsNode<FloatImm>();
+            if (float_imm) {
+              float imm = float_imm->value();
+              if (imm == 1.0f) {
+                return lhs;
+              } else if (imm == 2.0f) {
+                return lhs * lhs;
```

Comment:
Can we add a TODO to move this to the Codegen? It will be great to preserve the semantics as much as possible.

Reply:
I'm not sure you want to move this to CodeGen, as then we will need every backend to reimplement it.  If we had an IR->IR optimization phase, it would probably make sense there.

Reply:
Yes, a IR to IR optimization pass is where this should be.